js/reference.js

```diff
@@ -377,70 +372,52 @@ var ERMrest = (function(module) {
 
                 var defer = module._q.defer();
 
-                var uri = this._uri;
+                var uri = this._location.compactUri;
```

Comment:
We should not need `Reference.uri` anymore. I would think it should be completely replaced by `Reference.location` and if a client needs to get the uri, they do so through that interface.